### PR TITLE
Use passed name for leaky relu tensor op

### PR DESCRIPTION
**Pull request by jdgumz**
I was experimenting with different activation functions for the final layer of my graph recently when I noticed that the output graph was failing to save because it couldn't find a tensor by a name I had provided, e.g. (`my_final_tensor_op`).

It worked correctly with an activation function like `tf.nn.sigmoid`:
```
import tensorflow as tf
import numpy as np

sample_values = np.array([1.0, 2.0, 3.0], dtype=np.float64)
sigmoid_tensor = tf.nn.sigmoid(sample_values, name='my_final_tensor_op')
sigmoid_tensor.name
>>> 'my_final_tensor_op:0'
```

But for `tf.nn.leaky_relu` I noticed that a `/Maximum` gets appended to whatever `name` value is passed:
```
import tensorflow as tf
import numpy as np

sample_values = np.array([1.0, 2.0, 3.0], dtype=np.float64)
leaky_relu_tensor = tf.nn.leaky_relu(sample_values, name='my_final_tensor_op')
leaky_relu_tensor.name
>>> 'my_final_tensor_op/Maximum:0'
```

I suspect the reason why is that the `name` parameter [is not passed in the call](https://github.com/tensorflow/tensorflow/blob/f91bd2f8c4b263dd5460e4398b94ad4823ce7a18/tensorflow/python/ops/nn_ops.py#L1604) to the `math_ops.maximum` function:
```
return math_ops.maximum(alpha * features, features)
```

Compare this to the case of `tf.nn.sigmoid`, which [does pass in](https://github.com/tensorflow/tensorflow/blob/f91bd2f8c4b263dd5460e4398b94ad4823ce7a18/tensorflow/python/ops/math_ops.py#L2342) the `name` parameter into the function call that it returns:
```
return gen_math_ops.sigmoid(x, name=name)
```

This PR makes the change to have the `leaky_relu` function pass `name` to the `math_ops.maximum` function so that the desired name for the op carries down. I also added a unit test that addresses this specific functionality.

One potential issue that could come up is if there's a lot of existing code that expects the `/Maximum` string to be appended, such as in the case where no `name` is set and the tensor op's name becomes `LeakyRelu/Maximum:0`. If that's the case, I would at least like to change the method's documentation so that the caller is aware of the `/Maximum` string concatenation side effect.

### CODE REVIEW

**Comment by vrv** [comment]
I made one small change ("with ops.name_scope(...) as name:") based on the other functions in this file work, but otherwise, looks good.  You probably have to sign the CLA before we can accept this change.

**Comment by jdgumz** [comment]
I signed it!

**Review by rmlarsen — DISMISSED** [review]

**Comment by rmlarsen** [comment]
@vrv thanks for the review.

**Comment by jdgumz** [comment]
There was a small bug in the test - think I got confused between `nn` and `np`. 😝 

Anyhow, I corrected the function call so that it matches how it works in the other leaky relu tests.

**Comment by rmlarsen** [comment]
@jdgumz Please address the linter errors: https://source.cloud.google.com/results/invocations/bd5cab36-e615-4bb3-859e-0c6bd51d9869/log

**Comment by jdgumz** [comment]
@rmlarsen Done!

Looks like the pylint errors were:
```
FAIL: Found 2 non-whitelited pylint errors:
tensorflow/python/ops/nn_test.py:967: [C0301(line-too-long), ] Line too long (99/80)
tensorflow/python/ops/nn_test.py:969: [C0301(line-too-long), ] Line too long (81/80)
```

I shortened the lines in the test and ran pylint locally before pushing the latest commit. I didn't see the failures come up again, so hopefully it's resolved now.

**Review by vrv — APPROVED** [review]
Thanks!

**Comment by rmlarsen** [comment]
@jdgumz Thanks!

**Comment by jdgumz** [comment]
@rmlarsen 
@vrv 

I took a look at the `Windows CMake` build failure and it seems like it's under the `python/kernel_tests/boosted_trees` test coverage. I don't think my changes are related to the failure, but I wonder if perhaps my branch is behind and is missing a commit that addresses the test failure. Will I need to rebase the branch to perhaps fix the problem?

**Comment by rmlarsen** [comment]
@jdgumz Thanks. The failure seems unrelated to your change.
